src/methods/join.js

```diff
@@ -443,7 +443,7 @@ export function joinMaps<K, V>(
 // where the join is defined to be just m1[key] if m1[key] === m2[key] and
 // and abstract value with expression "joinCondition ? m1[key] : m2[key]" if not.
 export function joinBindings(realm: Realm, joinCondition: AbstractValue, m1: Bindings, m2: Bindings): Bindings {
-  function getAbstractValue(v1: void | Value, v2: void | Value): AbstractValue {
+  function getAbstractValue(v1: void | Value, v2: void | Value): Value {
```

Comment:
Why the change of return type? To make Flow happy? Don't see any other dataflow reason.

Reply:
Join calls AbstractValue. createFromConditionalOp which now can return a concrete value as a result of refinement and simplification.